### PR TITLE
DOC-8104: PR #104265 - util/log: timezone support for formats crdb-v1 and crdb-v2

### PR DESCRIPTION
**Pull request by florence-crl**
Addresses DOC-8104 DOC-8102 DOC-8103

(1) Updated Configure Logs page, added Set timezone section. (2) Updated cockroach debug merge-logs page, added required format option and considerations section.

### CODE REVIEW

File: src/current/v23.2/cockroach-debug-merge-logs.md [hunk 1]
```diff
@@ -39,44 +40,48 @@ Generate a debug zip file:
...
 
 {% include_cached copy-clipboard.html %}
 ~~~ shell
-cockroach debug merge-logs --redact debug/nodes/*/logs/*
+cockroach debug merge-logs debug/nodes/*/logs/* --format=crdb-v1 --redact
 ~~~
 
+## Consideration
+
+As of v23.2, logs can be configured to use a [timezone with formats `crdb-v1` or `crdb-v2`]({% link {{ page.version.version }}/configure-logs.md %}#set-timezone). This is a [backward-incompatible change](https://www.cockroachlabs.com/docs/releases/v23.2#v23-2-0-alpha-1-backward-incompatible-changes). With versions prior to v23.2, `cockroach debug merge-log` will return an error if run against logs with the timezone configured. `cockroach debug merge-log` requires v23.2 or later if run against logs with the timezone configured.
```
Comment (abarganier) [inline src/current/v23.2/cockroach-debug-merge-logs.md:83]: Small typo `cockroach debug merge-log` -> `cockroach debug merge-logs`. I think the same typo was in the release-note (oops!)

File: src/current/v23.2/configure-logs.md [hunk 2]
```diff
@@ -483,6 +483,31 @@ sinks:
       channels: {INFO: DEV, ERROR: OPS}
 ~~~
 
+### Set timezone
```
Comment (abarganier) [inline src/current/v23.2/configure-logs.md:486]: Just to clarify, will the datetime/timezone format-options for the json format be handled in a separate PR?

**Review by abarganier — COMMENTED** [review]
Looking good! A couple small comments, but otherwise LGTM

File: src/current/v23.2/cockroach-debug-merge-logs.md (hunk 1 above)
Comment (florence-crl) [reply src/current/v23.2/cockroach-debug-merge-logs.md]: corrected typo. Published release note was correct.

File: src/current/v23.2/configure-logs.md (hunk 2 above)
Comment (florence-crl) [reply src/current/v23.2/configure-logs.md:486]: Yes, the datetime/timezone format-options for the json format will be handled in a separate PR.

Comment (abarganier) [reply src/current/v23.2/configure-logs.md:486]: Ack, thanks

**Review by abarganier — APPROVED** [review]
LGTM from my end, thank you Florence!

File: src/current/v23.2/cockroach-debug-merge-logs.md [hunk 3]
```diff
@@ -39,44 +40,48 @@ Generate a debug zip file:
...
-$ unzip ./cockroach-data/logs/debug.zip
+unzip ./cockroach-data/logs/debug.zip
 ~~~
 
 Merge the logs in the debug folder:
 
 {% include_cached copy-clipboard.html %}
 ~~~ shell
-$ cockroach debug merge-logs debug/nodes/*/logs/*
+cockroach debug merge-logs debug/nodes/*/logs/* --format=crdb-v1
```
Comment (rmloveland) [inline src/current/v23.2/cockroach-debug-merge-logs.md:57]: I notice these examples use the `crdb-v1` format but https://www.cockroachlabs.com/docs/stable/log-formats calls it a "legacy format".  Are we still encouraging the use of `crdb-v1`?  What would happen if these commands used `crdb-v2`?  Is the upshot here that I as a user need to know/care which logging version was in use if I'm going to run `merge-logs` ?

File: src/current/v23.2/cockroach-debug-merge-logs.md [hunk 4]
```diff
@@ -39,44 +40,48 @@ Generate a debug zip file:
...
 
 You can redact sensitive information from the merged logs:
 
 {% include_cached copy-clipboard.html %}
 ~~~ shell
-cockroach debug merge-logs --redact debug/nodes/*/logs/*
+cockroach debug merge-logs debug/nodes/*/logs/* --format=crdb-v1 --redact
 ~~~
 
+## Consideration
```
Comment (rmloveland) [inline src/current/v23.2/cockroach-debug-merge-logs.md:81]: This section should be called 'Considerations' (plural) to match our other pages (e.g. https://www.cockroachlabs.com/docs/v23.1/import-into#considerations)

File: src/current/v23.2/cockroach-debug-merge-logs.md (hunk 3 above)
Comment (rmloveland) [reply src/current/v23.2/cockroach-debug-merge-logs.md:57]: Ah ok after reading this again, I see that this example is looking at the v1 logs created by `debug zip`

I'm kinda bummed that as a user I need to care about these different formats :-/ (but that is beyond the scope of this docs PR obv)

File: src/current/v23.2/configure-logs.md [hunk 5]
```diff
@@ -483,6 +483,31 @@ sinks:
...
+
+The log output formats [`crdb-v1`]({% link {{ page.version.version }}/log-formats.md %}#format-crdb-v1) and [`crdb-v2`]({% link {{ page.version.version }}/log-formats.md %}#format-crdb-v2) support the format option `timezone`. When specified, the corresponding timezone is used to produce the timestamp column. The value can be any [timezone name recognized by the Go standard library](https://github.com/arp242/tz/blob/bf333631bec4/list.go).
```
Comment (rmloveland) [inline src/current/v23.2/configure-logs.md:488]: This paragraph should link back to the 'Considerations' section on cockroach-debug-merge-logs.md because (If I'm understanding this correctly) once you set this setting it's a one-way ratchet, which will prevent you from ever doing `cockroach debug merge-logs` with a binary < v23.2.  Right?  Please correct me if I'm misunderstanding this!

File: src/current/v23.2/cockroach-debug-merge-logs.md [hunk 6]
```diff
@@ -24,10 +24,11 @@ $ cockroach debug merge-logs [log file directory] [flags]
 
 ## Flags
 
-Use the following flags to filter the `debug merge-logs` results for a specified regular expression or time range.
+The `debug merge-logs` subcommand supports the following flags:
 
 Flag | Description
 -----|-----------
+`--format` | **Required** The [format of the logs]({% link {{ page.version.version }}/log-formats.md %}) in the log file directory. The logs in a `.zip` generated using [`cockroach debug zip`]({% link {{ page.version.version }}/cockroach-debug-zip.md %}) are of type [`crdb-v1`]({% link {{ page.version.version }}/log-formats.md %}#format-crdb-v1).
```
Comment (rmloveland) [inline src/current/v23.2/cockroach-debug-merge-logs.md:31]: This tells me what is going to be in v1.  How do I know what logs are going to be in the v2 format?  

I guess as a user I should know whether my logs are v1 or v2 because I specified it in the [YAML logging config](https://www.cockroachlabs.com/docs/v23.2/configure-logs#yaml-payload) at start time right?  Sorry if these are dumb questions, just trying to make sure I understand

File: src/current/v23.2/cockroach-debug-merge-logs.md (hunk 3 above)
Comment (rmloveland) [reply src/current/v23.2/cockroach-debug-merge-logs.md:57]: OK I'm less bummed now that I know I'm the one configuring the formats via https://www.cockroachlabs.com/docs/v23.2/configure-logs#yaml-payload 

Sorry for the noise on my "learning journey" in this PR :-)

**Review by rmloveland — APPROVED** [review]
LGTM 

I do think we need the links to be bidirectional between these two pages, and we should name that one section 'Considerations' to match our other docs (see comments below)

Otherwise my other comments are just non-blocking suggestions and/or reflect my learning journey around logging (which I haven't really looked at before)

File: src/current/v23.2/cockroach-debug-merge-logs.md (hunk 6 above)
Comment (florence-crl) [reply src/current/v23.2/cockroach-debug-merge-logs.md:31]: To clarify,
- If you run `debug zip merge-logs` against the actual logs of a cockroach node you will use the format that you specified for the logs.
- If you run `debug zip merge-logs` against the logs unzipped from a generated `.zip` file, the format will be `crdb-v1` because `debug zip` converts the whatever the format of the actual logs into `crdb-v1`.
Does that make sense? We were surprised that this was the case.

File: src/current/v23.2/cockroach-debug-merge-logs.md (hunk 4 above)
Comment (florence-crl) [reply src/current/v23.2/cockroach-debug-merge-logs.md:81]: done

File: src/current/v23.2/cockroach-debug-merge-logs.md [hunk 7]
```diff
@@ -39,44 +40,48 @@ Generate a debug zip file:
...
 
 {% include_cached copy-clipboard.html %}
 ~~~ shell
-cockroach debug merge-logs --redact debug/nodes/*/logs/*
+cockroach debug merge-logs debug/nodes/*/logs/* --format=crdb-v1 --redact
 ~~~
 
+## Consideration
+
+As of v23.2, logs can be configured to use a [timezone with formats `crdb-v1` or `crdb-v2`]({% link {{ page.version.version }}/configure-logs.md %}#set-timezone). This is a [backward-incompatible change](https://www.cockroachlabs.com/docs/releases/v23.2#v23-2-0-alpha-1-backward-incompatible-changes). With versions prior to v23.2, `cockroach debug merge-logs` will return an error if run against logs with the timezone configured. `cockroach debug merge-logs` requires v23.2 or later if run against logs with the timezone configured.
```
[the remark this answers is not in the document]
Comment (florence-crl) [reply src/current/v23.2/cockroach-debug-merge-logs.md:83]: done

File: src/current/v23.2/configure-logs.md (hunk 5 above)
Comment (florence-crl) [reply src/current/v23.2/configure-logs.md:488]: Discussed with @rmloveland. We decided that customers don't need to know the details of `merge-logs` which is used by TSE team.

**Review by florence-crl — COMMENTED** [review]
TFTR!
